src/conditionalriskofkmins.py

```diff
@@ -0,0 +1,112 @@
+import numpy as np
+from scipy.stats import hypergeom
+from scipy.signal import fftconvolve
+import datetime as dt
+
+starttime = dt.datetime.now()
+
+''' This function aids in truncating distributions by finding levels l and u such that 
+    cdf(l) < .0000001 and 1 - cdf(u) < .0000001. (Here, cdf is used somewhat loosely because we do not require cdf(infinity) = 1,
+    although the distribution should sum "close enough" to 1 because .0000001 is absolute, not relative 
+    (i.e. a distribution that summed to .0000004 would result in only half the distribution being between l and u). 
+    
+    The purpose of this is to improve efficiency, since, for instance almost all of the hypergeometric distribution falls between a fraction of its range. 
+    This decreases the time it takes to iterate over the (meaningful parts of) distributions. '''
+def get_interval(dist):
+    length = len(dist)
+
+    lower_sum = 0
+    lower_endpoint = 0
+    for i in range(0, length):
+        lower_sum += dist[i]
+
+        # if adding the next value of the distribution would cause us to exceed the tolerance, break and return the lower level
+        if (lower_sum + dist[i + 1] > .000001):
+            lower_endpoint = i
+            break
+    
+    upper_sum = 0
+    upper_endpoint = length
+    for i in range(0, length):
+        upper_sum += dist[length - i - 1]
+
+        if (upper_sum + dist[length - i - 2] > .000001):
+            upper_endpoint = length - i - 1
+            break
+    
+    endpoints = [lower_endpoint, upper_endpoint]
+    return endpoints
+
+# N = total number of votes for the two candidates
```

Comment:
It is very helpful for testing and modularity to put code that actually does work in a `main()` function, called via the __main__ convention at the end of the file.